instrumentation/couchbase/couchbase-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/couchbase/v3_1/CouchbaseInstrumentationModule.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.couchbase.v3_1;
+
+import static io.opentelemetry.javaagent.tooling.bytebuddy.matcher.ClassLoaderMatcher.hasClassesNamed;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.tooling.InstrumentationModule;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.Collections;
+import java.util.List;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(InstrumentationModule.class)
+public class CouchbaseInstrumentationModule extends InstrumentationModule {
+  public CouchbaseInstrumentationModule() {
+    super("couchbase", "couchbase-3.1");
+  }
+
+  @Override
+  protected String[] additionalHelperClassNames() {
+    return new String[] {
+      "com.couchbase.client.tracing.opentelemetry.OpenTelemetryRequestSpan",
+      "com.couchbase.client.tracing.opentelemetry.OpenTelemetryRequestTracer"
+    };
+  }
```

Comment:
This is interesting: if we're going to include external library instrumentations in the javaagent (and we're probably going to do that, since it seems to be a good way to make the agent a bit slimmer & simpler) then we need to improve muzzle a bit.
Right now muzzle not only does not collect helper classes from 3rd party packages, it also collects no references whatsoever. Which means that we actually don't have the muzzle check both in build time and runtime.
It should be possible to define the set of packages for each module, by default it'd be same as `InstrumentationClassPredicate` - I'm going to open an issue for that.

Reply:
Turns out there already was one: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1395
I've just added my 3 cents there